### DISCUSSION TITLE
Is curl_global_cleanup() really necessary?

### DISCUSSION THREAD

**Discussion by philbond007** [curl/curl#11919] [Q&A]
Hi all,

I'm using libcurl from a library I want to offer to clients.
Now I want to statically link-in libcurl.

So evidently I will have to curl_global_init() once for the lifetime of the client .exe which will have also statically linked-in my libcode(which has statically linked libcurl).

Now I wonder if it's relevant to call curl_global_cleanup(), because when there will be no more libcurl-functionality needed, this means the process of the client .exe will have disappeared, together with its complete address space, so what is there to cleanup?

Tx.

**Comment by awesomekosm** [comment]
Same question here.
Also, if curl_global_cleanup is ever called and then for some reason curl_global_init is called again with curl_multi_init and curl_easy_init, there is an obscure error.
```
 mbedtls_ctr_drbg_seed returned (-0x0034) CTR_DRBG - The entropy source failed
```

**Reply by bagder** [reply]
That sounds like (an mbedtls specific) bug

**Reply by jay** [reply]
@awesomekosm it might be a bug in curl. Can you please try #13071

**Reply by awesomekosm** [reply]
Thanks @jay, it works! I've tested your branch on mac clang version 15.0, I can do `curl_global_cleanup` and then `curl_global_init` again followed by successful `curl_multi_perform`.
